plugin/web/extensions/sidebar/core/components/mobile/hooks/index.ts

```diff
@@ -18,20 +18,41 @@ export default () => {
   const [backendURL, setBackendURL] = useState<string>();
   const [backendProjectName, setBackendProjectName] = useState<string>();
 
+  const [isCustomProject, setIsCustomProject] = useState<boolean>(false);
+  const [customCatalogURL, setCustomCatalogURL] = useState<string>();
+  const [customCatalogProjectName, setCustomCatalogProjectName] = useState<string>();
+  const [customReearthURL, setCustomReearthURL] = useState<string>();
+  const [customBackendURL, setCustomBackendURL] = useState<string>();
+  const [customBackendProjectName, setCustomBackendProjectName] = useState<string>();
+
   const [customProjectName, setCustomProjectName] = useState<string>();
   const [customLogo, setCustomLogo] = useState<string>();
 
   const [fieldTemplates, setFieldTemplates] = useState<Template[]>([]);
   const [infoboxTemplates, setInfoboxTemplates] = useState<Template[]>([]);
 
   const [searchTerm, setSearchTerm] = useState("");
+  const [customSearchTerm, setCustomSearchTerm] = useState("");
 
   useEffect(() => {
     (async () => {
-      if (!backendURL || !backendProjectName) return;
-      const res = await fetch(`${backendURL}/sidebar/${backendProjectName}/templates`);
-      if (res.status !== 200) return;
-      const resData = await res.json();
+      let resData: Template[] = [];
+      if (backendURL && backendProjectName) {
+        const res = await fetch(`${backendURL}/sidebar/${backendProjectName}/templates`);
+        if (res.status === 200) {
+          const plateauResData = (await res.json()) as Template[];
+          resData = plateauResData.map(t => ({ ...t, dataSource: "plateau" }));
+        }
+      }
+      if (isCustomProject) {
+        const res = await fetch(
+          `${customBackendURL}/sidebar/${customBackendProjectName}/templates`,
+        );
+        if (res.status === 200) {
+          const customResData = (await res.json()) as Template[];
+          resData = resData.concat(customResData.map(t => ({ ...t, dataSource: "custom" })));
```

Comment:
Can you move adding dataSource in to the api code? If you do it here, then we are processing the datacatalog in multiple places, which will double the processing time)